main/src/library/Random.flix

```diff
@@ -84,24 +86,29 @@ mod Random {
     ///
     /// In other words, re-interprets the `Random` effect using seeded deterministic randomness.
     ///
-    pub def handleWithSeed(seed: Int64, f: a -> b \ ef): (a -> b \ (ef - Random) + {NonDet, IO}) \ {NonDet, IO} =
-        let rnd = new JRandom(seed); // Reuse Random instance.
-        x -> run {
-            f(x)
-        } with Random {
-            def randomBool(k)       = k(rnd.nextBoolean())
-            def randomFloat32(k)    = k(rnd.nextFloat())
-            def randomFloat64(k)    = k(rnd.nextDouble())
-            def randomInt32(k)      = k(rnd.nextInt())
-            def randomInt64(k)      = k(rnd.nextLong())
-            def randomGaussian(k)   = k(rnd.nextGaussian())
+    pub def handleWithSeed(seed: Int64, f: a -> b \ ef): a -> b \ ef - Random = {
+        // Given a specifc seed, JRandom is just a mutable iterator and therefore we type it as such.
+        x -> region local {
+            let _ = local; // avoid redundancy check
+            let rnd = unchecked_cast (new JRandom(seed) as _ \ local);
+            run {
+                f(x)
+            } with Random {
+                def randomBool(k)       = k(unchecked_cast (rnd.nextBoolean() as _ \ local))
+                def randomFloat32(k)    = k(unchecked_cast (rnd.nextFloat() as _ \ local))
+                def randomFloat64(k)    = k(unchecked_cast (rnd.nextDouble() as _ \ local))
+                def randomInt32(k)      = k(unchecked_cast (rnd.nextInt() as _ \ local))
+                def randomInt64(k)      = k(unchecked_cast (rnd.nextLong() as _ \ local))
+                def randomGaussian(k)   = k(unchecked_cast (rnd.nextGaussian() as _ \ local))
```

Comment:
I could've used `unsafely {NonDet, IO} run ..` but i also need to add `local` which we don't have nice syntax for